pkg/gameservers/controller.go

```diff
@@ -573,20 +573,22 @@ func (c *Controller) createGameServerPod(gs *agonesv1.GameServer) (*agonesv1.Gam
 
 	c.loggerForGameServer(gs).WithField("pod", pod).Debug("Creating Pod for GameServer")
 	pod, err = c.podGetter.Pods(gs.ObjectMeta.Namespace).Create(pod)
-	if k8serrors.IsAlreadyExists(err) {
-		c.recorder.Event(gs, corev1.EventTypeNormal, string(gs.Status.State), "Pod already exists, reused")
-		return gs, nil
-	}
 	if err != nil {
-		if k8serrors.IsInvalid(err) {
+		switch {
+		case k8serrors.IsAlreadyExists(err):
+			c.recorder.Event(gs, corev1.EventTypeNormal, string(gs.Status.State), "Pod already exists, reused")
+			return gs, nil
+		case k8serrors.IsInvalid(err):
 			c.loggerForGameServer(gs).WithField("pod", pod).Errorf("Pod created is invalid")
 			gs, err = c.moveToErrorState(gs, err.Error())
 			return gs, err
-		} else if k8serrors.IsForbidden(err) {
+		case k8serrors.IsForbidden(err):
 			c.loggerForGameServer(gs).WithField("pod", pod).Errorf("Pod created is forbidden")
 			gs, err = c.moveToErrorState(gs, err.Error())
 			return gs, err
 		}
+		c.loggerForGameServer(gs).WithField("pod", pod).WithError(err)
+		c.recorder.Eventf(gs, corev1.EventTypeWarning, string(gs.Status.State), "error creating Pod for GameServer %s", gs.Name)
```

Comment:
Should this be a `default:` state of the switch, rather than sit outside the switch?

Since `moveToErrorState` will create an event, then we'll get 2 events with the same Error warning if the errors types is `AlreadyExists`, `Invalid` or `Forbidden` ?

Reply:
Yes, that is true